src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1002,33 +1002,34 @@ public override string GetDocumentationCommentXml(CultureInfo preferredCulture =
 
         internal override DiagnosticInfo GetUseSiteDiagnostic()
         {
+            DiagnosticInfo result = null;
+
             if (!_packedFlags.IsUseSiteDiagnosticPopulated)
             {
-                DiagnosticInfo result = null;
                 CalculateUseSiteDiagnostic(ref result);
                 EnsureTypeParametersAreLoaded(ref result);
-                return InitializeUseSiteDiagnostic(result);
-            }
-
-            var uncommonFields = _uncommonFields;
-            if (uncommonFields == null)
```

Comment:
Took a look. We won't be loosing an optimization. `CSDiagnosticInfo.IsEmpty` will return `false` for `null` results. Therefore, it won't be initialized. I believe the changed code is simpler to read now.

---
In reply to: [119753756](https://github.com/dotnet/roslyn/pull/19658#discussion_r119753756) [](ancestors = 119753756)